### PR TITLE
Adds QGB related ADRs

### PR DESCRIPTION
**Pull request by SweeXordious**
Closes: https://github.com/celestiaorg/celestia-app/issues/217

### CODE REVIEW

**Comment by SweeXordious** [comment]
@evan-forbes Please take a look on this if it is in the right direction. And, if you have any idea what needs to be added to this :D Thanks a lot

File: docs/architecture/ADR-002-QGB-Overview.md
Comment (evan-forbes) [inline docs/architecture/ADR-002-QGB-Overview.md:12]: This is an important statement to discuss, but I think starting off with something more fundamental to why we could do this in the first place would be useful. Rewriting the relayer and the orchestrator was done because our relayer can be designed to be trust minimized, where a two way bridge requires would require each validator to run a trusted relayer. Since we had to rewrite them and they were simplified, we decided to move things to the app. Either in this section or another, we could also compare and contract the consequences of moving to the app.

We might also want to discuss how we are using a similar design to a portion of the normal gravity bridge's application integration. Technically, we don't even need the additional state for the MVP, we only need the state for slashing, but since we are planning on doing slashing in the future, the state changes make it trivial for the relayer to find the data needed to relay to the evm chain.

**Review by evan-forbes — COMMENTED** [review]
This is a great start! :+1:  Originally I had only imagined a single ADR, but I like having three smaller ones. Left some comments

**Comment by SweeXordious** [comment]
@evan-forbes Please take a look when you have time again

File: docs/architecture/ADR-002-QGB-Overview.md
Comment (evan-forbes) [inline docs/architecture/ADR-002-QGB-Overview.md:9]: as discussed in the sync, we want to focus on the app integration portion here. I'm still thinking on exactly what needs to be covered, because technically we don't need any of the app integration until slashing. If we didn't need to slash validators, then we would do everything off chain to save state bloat.

File: docs/architecture/ADR-003-QGB-ValSet.md
Comment (evan-forbes) [inline docs/architecture/ADR-003-QGB-ValSet.md:28]: A++

**Review by evan-forbes — COMMENTED** [review]
thinking on this a bit more, we might just want to not include the overview ADR yet. We only need to add state to the app for the qgb when we have slashing, if we weren't planning on doing that, then we would do everything off chain. Unless we want to talk about that, I actually think its okay to get rid of the overview portion. Perhaps we can add in some grander overview later? over perhaps in the qgb repo. Still not sure yet.

The other two ADRs are good tho, and we should definitely include them

File: docs/architecture/ADR-002-QGB-Overview.md
Comment (adlerjohn) [inline docs/architecture/ADR-002-QGB-Overview.md:19]: Don't manually add newlines for markdown files, except maybe after sentences if you want each sentence on a different line for git diffs.
```suggestion
The QGB allows Celestia block header data roots to be relayed in one direction, from Celestia to an EVM chain. It does not support bridging assets such as fungible or non-fungible tokens directly, and cannot send messages from the EVM chain back to Celestia.
```

File: docs/architecture/ADR-003-QGB-ValSet.md
Comment (adlerjohn) [inline docs/architecture/ADR-003-QGB-ValSet.md:26]: Link to ADR?

File: docs/architecture/ADR-003-QGB-ValSet.md
Comment (adlerjohn) [inline docs/architecture/ADR-003-QGB-ValSet.md:30]: Do we need to submit a "new" validator set within the unbonding window even if it hasn't changed? Otherwise the bridge contract doesn't know if the bridge should shut down due to no updates within the unbonding window, or if there simply were no updates.

File: docs/architecture/ADR-003-QGB-ValSet.md
Comment (adlerjohn) [inline docs/architecture/ADR-003-QGB-ValSet.md:45]: Why `string` and not `bytes`?

File: docs/architecture/ADR-003-QGB-ValSet.md
Comment (adlerjohn) [inline docs/architecture/ADR-003-QGB-ValSet.md:59]: Alignment here is on types, while alignment about is on field number. Make uniform.

File: docs/architecture/ADR-003-QGB-ValSet.md
Comment (adlerjohn) [inline docs/architecture/ADR-003-QGB-ValSet.md:65]: These should honestly be comments in the protobuf, so that when it's copy-pasted the comments are there.

File: docs/architecture/ADR-003-QGB-ValSet.md
Comment (SweeXordious) [reply docs/architecture/ADR-003-QGB-ValSet.md]: If I understand right, https://github.com/SweeXordious/celestia-app/blob/e4760741315840395c589d44414a185c1f641084/x/qgb/abci.go#L42
we will always post valsets requests even without it being changed. @evan-forbes Correct me if I am wrong please

File: docs/architecture/ADR-003-QGB-ValSet.md
Comment (SweeXordious) [reply docs/architecture/ADR-003-QGB-ValSet.md]: Because we will be submitting `0x...` hex encoded addresses

File: docs/architecture/ADR-003-QGB-ValSet.md
Comment (SweeXordious) [reply docs/architecture/ADR-003-QGB-ValSet.md]: Will put these in proto files also, then

File: docs/architecture/ADR-003-QGB-ValSet.md
Comment (SweeXordious) [reply docs/architecture/ADR-003-QGB-ValSet.md]: check now please if formatting is fixed ?

File: docs/architecture/ADR-003-QGB-ValSet.md
Comment (SweeXordious) [reply docs/architecture/ADR-003-QGB-ValSet.md]: We don't have an ADR in QGB, no ? I only found tracking issues. And these;
https://github.com/celestiaorg/quantum-gravity-bridge/issues/20
https://github.com/celestiaorg/quantum-gravity-bridge/issues/4
Do you mean one of them ?

File: docs/architecture/ADR-003-QGB-ValSet.md
Comment (evan-forbes) [reply docs/architecture/ADR-003-QGB-ValSet.md]: > Do we need to submit a "new" validator set within the unbonding window even if it hasn't changed? Otherwise the bridge contract doesn't know if the bridge should shut down due to no updates within the unbonding window, or if there simply were no updates.

it looks like it should work, but we don't have a test specifically for this. I'll write this down as a test case, and then @SweeXordious and I can plan around it today

File: docs/architecture/ADR-003-QGB-ValSet.md
Comment (SweeXordious) [reply docs/architecture/ADR-003-QGB-ValSet.md]: maybe:
```
Then, using `EndBlock`, the protocol makes a `ValsetRequest`...
```
?

**Review by evan-forbes — APPROVED** [review]
nice! all the documentation in the proto files was a great idea/addition.

:+1: :+1: LGTM

**Comment by SweeXordious** [comment]
@evan-forbes Can I merge then?

File: docs/architecture/ADR-003-QGB-ValSet.md
Comment (evan-forbes) [reply docs/architecture/ADR-003-QGB-ValSet.md]: I think John was referring simply to the other ADR included in this PR. We can just link to this that file in this PR I think

**Comment by evan-forbes** [comment]
might want to wait on @adlerjohn approval, but all my nits were covered I think

**Review by adlerjohn — APPROVED** [review]

**Pull request merged**
